crates/engine/tree/src/tree/mod.rs

```diff
@@ -2279,7 +2291,20 @@ where
         // Extend with block we are validating root for.
         input.append_ref(hashed_state);
 
-        Ok(ParallelStateRoot::new(consistent_view, input).incremental_root_with_updates()?)
+        let (tx, mut rx) = oneshot::channel();
+        let blocking_task_pool = self.blocking_task_pool.clone();
+
+        self.state_root_task_spawner.spawn_critical_blocking(
+            "state root task",
+            Box::pin(async move {
+                let res = AsyncStateRoot::new(consistent_view, blocking_task_pool, input)
+                    .incremental_root_with_updates()
+                    .await;
+                let _ = tx.send(res);
+            }),
+        );
+
+        rx.try_recv()?
```

Comment:
must be blocking receive instead
```suggestion
        rx.blocking_recv()?
```